src/viur/core/bones/uri.py

```diff
@@ -0,0 +1,192 @@
+import fnmatch
+from . import BaseBone, ReadFromClientError, ReadFromClientErrorSeverity
+from urllib.parse import urlparse, urlunparse
+import typing as t
+from collections.abc import Iterable
+from collections import namedtuple
+
+
+class URIBone(BaseBone):
+    """
+       The URIBone is used for storing URI and URL.
+
+       :param accepted_protocols: The accepted protocols can be set to allow only the provide protocols.
+       :param accepted_ports The accepted ports can be set to allow only the provide ports.
+       ..  code-block:: python
+            # Example
+            URIBone(accepted_ports=1)
+            URIBone(accepted_ports="2")
+            URIBone(accepted_ports="1-4")
+            URIBone(accepted_ports=[1,"2","4-10"])
```

Comment:
Sorry, I'm a tuple fan
```suggestion
            URIBone(accepted_ports=(1,"2","4-10"))
```